### PR TITLE
[#ICC-87] New Third party get message and attachments endpoints definition

### PR DESCRIPTION
**Pull request by gquadrati**
This PR depends on [io-fn-commons #ICC-87](https://github.com/pagopa/io-functions-commons/pull/275)

#### List of Changes

- add new `getThirdPartyMessage` and `getThirdPartyMessageAttachment` openapi definitions
- add new `getThirdPartyMessage` and `getThirdPartyMessageAttachment` in MessageController, returning `501 - Not Implemented`

#### Motivation and Context

Expose new endpoint for retrieving Third Party message details and attachments

#### How Has This Been Tested?

io-mock

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/utils/responses.ts [hunk 1]
```diff
@@ -187,3 +187,27 @@ export const wrapValidationWithInternalError: <A>(
...
+
+/**
+ * Interface for NotImplemented error response.
+ */
+export interface IResponseErrorNotImplemented
+  extends IResponse<"IResponseErrorNotImplemented"> {
+  readonly detail: string;
+}
+/**
+ * Returns an unauthorized error response with status code 501.
```
Comment (fabriziopapi) [inline src/utils/responses.ts:199]:
```suggestion
 * Returns a Not Implemented error response with status code 501.
```

**Review by fabriziopapi — CHANGES_REQUESTED** [review]

**Comment by fabriziopapi** [comment]
LGTM appena modificati i puntamenti per la fn-commons

**Comment by gquadrati** [comment]
> LGTM appena modificati i puntamenti per la fn-commons

I've updated fn-commons refs, but I need an approval form merging it

**Review by fabriziopapi — APPROVED** [review]

File: api_backend.yaml [hunk 2]
```diff
@@ -394,6 +394,112 @@ paths:
           description: Too Many Requests
         "500":
           description: Internal Server Error
+  "/third-party-messages/{id}":
+    x-swagger-router-controller: MessagesController
+    get:
+      operationId: getThirdPartyMessage
+      summary: Eetrieve Third Party message
```
Comment (gquadrati) [inline api_backend.yaml:401]:
```suggestion
      summary: Retrieve Third Party message
```

**Pull request merged**